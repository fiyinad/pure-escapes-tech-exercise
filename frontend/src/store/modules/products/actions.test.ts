import {
    getAllAction,
    getSingleAction,
    getAllRequestAction,
    getAllFailureAction,
    getAllSuccessAction
} from "./actions";

describe('Products actions', () => {
    it('GetAll action', () => {
        const payload = [
            { "id": 1, "name": "Soup", "customerPrice": 199, "cost": 186 },
            { "id": 2, "name": "Bread", "customerPrice": 87, "cost": 21 },
            { "id": 3, "name": "Cheese", "customerPrice": 275, "cost": 234 },
            { "id": 4, "name": "Milk", "customerPrice": 67, "cost": 61 }];
        expect(getAllAction(payload)).toMatchSnapshot();
    });

    it('GetSingle action', () => {
        const payload = { "id": 1, "name": "Soup", "customerPrice": 199, "cost": 186 };
        expect(getSingleAction(payload)).toMatchSnapshot();
    });

    it('Get ALL request action', () => {
        expect(getAllRequestAction()).toMatchSnapshot();
    });

    it('GET ALL success action', () => {
        expect(getAllSuccessAction()).toMatchSnapshot();
    });

    it('GET ALL failure action', () => {
        expect(getAllFailureAction()).toMatchSnapshot();
    });
});
