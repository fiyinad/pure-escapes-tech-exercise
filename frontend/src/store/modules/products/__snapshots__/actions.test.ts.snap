// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products actions GET ALL failure action 1`] = `
Object {
  "type": "GET_ALL_FAILURE",
}
`;

exports[`Products actions GET ALL success action 1`] = `
Object {
  "type": "GET_ALL_SUCCESS",
}
`;

exports[`Products actions Get ALL request action 1`] = `
Object {
  "type": "GET_ALL_REQUEST",
}
`;

exports[`Products actions GetAll action 1`] = `
Object {
  "payload": Array [
    Object {
      "cost": 186,
      "customerPrice": 199,
      "id": 1,
      "name": "Soup",
    },
    Object {
      "cost": 21,
      "customerPrice": 87,
      "id": 2,
      "name": "Bread",
    },
    Object {
      "cost": 234,
      "customerPrice": 275,
      "id": 3,
      "name": "Cheese",
    },
    Object {
      "cost": 61,
      "customerPrice": 67,
      "id": 4,
      "name": "Milk",
    },
  ],
  "type": "GET_ALL",
}
`;

exports[`Products actions GetSingle action 1`] = `
Object {
  "payload": Object {
    "cost": 186,
    "customerPrice": 199,
    "id": 1,
    "name": "Soup",
  },
  "type": "GET_SINGLE",
}
`;
